media/com_fabrik/js/tipsBootStrapMock.js

```diff
@@ -194,10 +195,18 @@ define(['jquery', 'fab/fabrik'], function (jQuery, Fabrik) {
 
                     actualWidth = $tip[0].offsetWidth;
                     actualHeight = $tip[0].offsetHeight;
-
-                    switch (inside ? placement.split(' ')[1] : placement) {
+                    var maxwidth = this.options.maxwidth;
+                    var xpos = parseInt(window.fabrikTipXpos);
```

Comment:
What is fabriktopXpos? There is no reference to it being set anywhere here or in Fabrik.

Since xpos is then used a lot at line 248, I can't understand how this works at all.

Reply:
To answer your prior concern - the code is already configured to just use the center position if the cursorX position is outside the breadth (width - ie. left or right edge) of the popover - which might happen with a narrow popover above or below a wide label.

I played with this code for 3 days before posting - so I'm pretty sure all the bases are covered - if you get my baseball analogy.

I explain where window.fabrikXpos comes from in the initial comment. The code is just below the screen snips.  I'm assuming that code will be in the form_xx.js, detals_xx.js, or list_xx.js file. And if it's not, no harm done - the arrow will just work like it does now. Or we can put it right in the tipsBootStrapMock.js file itself.  I just didn't want to force my preferences on anyone.

Reply:
This is not about forcing preferences on anyone - this is about improving this for everyone if it makes sense to do so. Having tooltips half off the screen makes no sense, so this PR does make sense to be global.

So, you need to code this to work for everyone without need for them to do anything.

Reply:
I made this change  - so it no longer requires window.fabriktopXpos to be initialized elsewhere.